### PR TITLE
NAT : Update the CRM used counters for SNAT and DNAT entries

### PR DESCRIPTION
**Pull request by AkhileshSamineni**
Issue : CRM used counters are not getting updated for SNAT and DNAT entries

Steps to recreate:
Add a static NAT entry and verify the CRM counters
```
root@sonic:/home/admin# config nat feature enable
root@sonic:/home/admin# config interface ip add Ethernet9 12.12.0.1/24
root@sonic:/home/admin# config interface ip add Ethernet11 125.56.90.12/24
root@sonic:/home/admin# config nat add interface Ethernet11 -nat_zone 1
root@sonic:/home/admin# 
root@sonic:/home/admin# config nat add static basic 125.56.90.8 12.12.0.2
root@sonic:/home/admin# show nat translations 

Static NAT Entries         ..................... 2
Static NAPT Entries        ..................... 0
Dynamic NAT Entries        ..................... 0
Dynamic NAPT Entries       ..................... 0
Static Twice NAT Entries   ..................... 0
Static Twice NAPT Entries  ..................... 0
Dynamic Twice NAT Entries  ..................... 0
Dynamic Twice NAPT Entries ..................... 0
Total SNAT/SNAPT Entries   ..................... 1
Total DNAT/DNAPT Entries   ..................... 1
Total Entries              ..................... 2

Protocol    Source     Destination    Translated Source    Translated Destination
----------  ---------  -------------  -------------------  ------------------------
all         12.12.0.2  ---            125.56.90.8          ---
all         ---        125.56.90.8    ---                  12.12.0.2

root@sonic:/home/admin#

=============After polling interval of 300 seconds ========

root@sonic:/home/admin# crm show resources snat

Resource Name    Used Count   Available Count
--------------- ------------ -----------------
snat_entry                0             1024

root@sonic:/home/admin#

root@sonic:/home/admin# crm show resources dnat

Resource Name    Used Count    Available Count
--------------- ------------ -----------------
dnat_entry                0             1024

root@sonic:/home/admin#
```
Fix: Increment/Decrement the crm used counters for snat/dnat entries when entry is created/deleted.

Repeated the same steps to add static nat entry like above and verified the crm counters.
```
root@sonic:/home/admin# crm show resources dnat

Resource Name     Used Count   Available Count
--------------- ------------ -----------------
dnat_entry                1             1023

root@sonic:/home/admin# crm show resources snat

Resource Name     Used Count   Available Count
--------------- ------------ -----------------
snat_entry                1             1023

root@sonic:/home/admin#
```
Signed-off-by: Akhilesh Samineni <akhilesh.samineni@broadcom.com>

### CODE REVIEW

**Comment by AkhileshSamineni** [comment]
@rlhui @arlakshm Please review the changes.

**Comment by AkhileshSamineni** [comment]
@arlakshm  Please review and merge the changes.

**Comment by arlakshm** [comment]
@kperumalbfn, can you please help review this change

**Comment by yxieca** [comment]
@AkhileshSamineni does this PR address https://github.com/Azure/sonic-buildimage/issues/7126?

**Comment by AkhileshSamineni** [comment]
> @AkhileshSamineni does this PR address [Azure/sonic-buildimage#7126](https://github.com/Azure/sonic-buildimage/issues/7126)?

@yxieca I don't think this PR changes will address the issue.

File: orchagent/natorch.cpp [hunk 1]
```diff
@@ -1720,6 +1729,7 @@ bool NatOrch::removeHwSnaptEntry(const NaptEntryKey &keyEntry)
     }
     deleteNaptCounters(keyEntry.prototype.c_str(), keyEntry.ip_address, keyEntry.l4_port);
     m_naptEntries.erase(keyEntry);
+    gCrmOrch->decCrmResUsedCounter(CrmResourceType::CRM_SNAT_ENTRY);
```
Comment (kperumalbfn) [inline orchagent/natorch.cpp:1732]: Current Sonic CrmOrch code doesn't have support for double_nat. When we add that support, we can update crm counters.

**Review by kperumalbfn — APPROVED** [review]

File: orchagent/natorch.cpp (hunk 1 above)
Comment (AkhileshSamineni) [reply orchagent/natorch.cpp:1732]: Yes, not yet supporting the double nat.

**Review by lguohan — APPROVED** [review]

**Pull request merged**